web/api/formatters/charts2json.c

```diff
@@ -8,30 +8,28 @@ const char* get_release_channel() {
     static int use_stable = -1;
 
     if (use_stable == -1) {
-		char filename[FILENAME_MAX + 1];
+        char filename[FILENAME_MAX + 1];
         snprintfz(filename, FILENAME_MAX, "%s/.environment", netdata_configured_user_config_dir);
         procfile *ff = procfile_open(filename, "=", PROCFILE_FLAG_DEFAULT);
-        if(!ff) {
-            use_stable=1;
-        } else {
+        if (ff) {
             procfile_set_quotes(ff, "'\"");
             ff = procfile_readall(ff);
-            if(!ff) {
-                use_stable=1;
-            } else {
+            if (ff) {
                 unsigned int i;
-                for(i = 0; i < procfile_lines(ff); i++) {
-                    if (!procfile_linewords(ff, i)) continue;
-
-                    if (!strcmp(procfile_lineword(ff, i, 0), "RELEASE_CHANNEL") && !strcmp(procfile_lineword(ff, i, 1), "stable")) {
+                for (i = 0; i < procfile_lines(ff); i++) {
+                    if (!procfile_linewords(ff, i))
+                        continue;
+                    if (!strcmp(procfile_lineword(ff, i, 0), "RELEASE_CHANNEL") &&
+                        !strcmp(procfile_lineword(ff, i, 1), "stable")) {
```

Comment:
we need to check for 'nightly' here too, so we don't fall back to parsing the program version if the file exists and RELEASE_CHANNEL is nightly

Reply:
Good point